sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -82,36 +84,155 @@ private DoFnSignatures() {}
     return signature;
   }
 
+  /**
+   * The global context for the class, for use in analysis.
+   *
+   * <p>It contains much of the information that eventually becomes part of the {@link
+   * DoFnSignature}, but in an intermediate state.
+   */
+  @VisibleForTesting
+  static class GlobalAnalysisContext {
+
+    private final Map<String, StateDeclaration> stateDeclarations = new HashMap<>();
```

Comment:
Since this is essentially a private class, I'm not sure that you need to encapsulate these maps at all. I'd probably remove lines 100-134 and have callers access the variables directly; likewise in MethodAnalysisContext. But up to you.